src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -707,9 +710,11 @@ void createEventsTransaction(final List<O> ops) {
 				// create the batch event stream writer if necessary
 				val txnEvtWriterCache = threadLocalTxnEvtWriterCache.get();
 				val batchEvtWriter = txnEvtWriterCache.computeIfAbsent(
-								clientFactory,
-								clientFactory_ -> clientFactory_.createTransactionalEventWriter(
-												streamName, evtSerializer, evtWriterConfig));
+					clientFactory,
+					clientFactory_ -> clientFactory_.createTransactionalEventWriter(
+						UUID.randomUUID().toString(), streamName, evtSerializer, evtWriterConfig
```

Comment:
I've seen both Long.toString(System.nanoTime()) and now UUID.randomUUID().toString() to create a unique ID for an instance. Is there a general guideline when to use which?

Reply:
This call uses `UUID.randomUUID().toString()` internally if the arg is omitted, so I just left this the same